src/container/srv_container.c

```diff
@@ -3519,6 +3519,21 @@ upgrade_cont_cb(daos_handle_t ih, d_iov_t *key, d_iov_t *val, void *varg)
 	if (rc)
 		goto out;
 
+	global_ver = DS_POOL_GLOBAL_VERSION;
```

Comment:
this is a question about the code above on line 3510. It is possible I may be missing the overall context of how this code is run in the container upgrade procedure.

If the rdb_tx_lookup() returns -DER_NONEXIST, then the test of global_var on that line 3510 will be based on an uninitialized value is that right? Should global_var be assigned to 0 initially at the declaration?

Reply:
You are right.